cosmo_tester/test_suites/ha/ha_cluster_scenarios_test.py

```diff
@@ -71,31 +72,40 @@ def cluster(
 
 
 @pytest.fixture(scope='function')
-def hello_world(cfy, cluster, attributes, ssh_key, tmpdir, logger):
-    hw = HelloWorldExample(
-        cfy, cluster.managers[0], attributes, ssh_key, logger, tmpdir)
-    hw.blueprint_file = 'openstack-blueprint.yaml'
-    hw.inputs.update({
-        'agent_user': attributes.centos7_username,
-        'image': attributes.centos7_image_name,
-    })
-
-    yield hw
-    if hw.cleanup_required:
-        logger.info('Hello world cleanup required..')
-        cluster.managers[0].use()
-        hw.cleanup()
-
-
-def test_data_replication(cfy, cluster, hello_world,
+def hello_worlds(cfy, cluster, attributes, ssh_key, tmpdir, logger):
+    # Pick a manager to operate on, and trust the cluster to work with us
+    manager = cluster.managers[0]
+
+    hws = []
+    for i in range(0, 2):
+        tenant = get_test_tenant('clusterhello{num}'.format(num=i),
+                                 manager, cfy)
+        hw = HelloWorldExample(
+            cfy, manager, attributes, ssh_key, logger, tmpdir,
+            tenant=tenant, suffix=str(i),
+        )
+        hw.blueprint_file = 'openstack-blueprint.yaml'
+        hw.inputs.update({
+            'agent_user': attributes.centos_7_username,
+            'image': attributes.centos_7_image_name,
+        })
+        hws.append(hw)
+
+    yield hws
+    for hw in hws:
+        if hw.cleanup_required:
```

Comment:
Why wasn't this `if` necessary in `get_hello_worlds`?

Reply:
This was just for the output- it shouldn't actually be required to make the tests work I think. Worth removing or not?

Reply:
Wait the cleanup isn't required, or the `cleanup_required` flag isn't?

Reply:
The check technically isn't required, since cleanup() checks whether that flag is set. In fact, it appears that the hello world actually logs this anyway... why on earth did I put that in there? I'll remove it.